{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -155,52 +245,92 @@ def generate_secret_key():
+
+
+@task
+@remote
+def fetch_db(c, destination="."):
     """
-    Create the basic directory structure on the remote server.
+    Dump the database on the remote host and retrieve it locally.
+
+    The destination parameter controls where the dump should be stored locally.
     """
-    run("mkdir -p %s" % env.root)
+    dump_path = c.conn.dump_db("~")
+    filename = os.path.basename(dump_path)
+
+    subprocess.run(
+        [
+            "scp",
+            "-P",
+            str(c.conn.port),
+            "{user}@{host}:{directory}".format(
+                user=c.conn.user, host=c.conn.host, directory=dump_path
+            ),
+            destination,
+        ]
+    )
+    c.conn.run("rm %s" % dump_path)
 
-    with cd(env.root):
-        run("mkdir -p static backups")
-        run("python3 -m venv venv")
+    return os.path.join(destination, filename)
 
 
 @task
-def sync_settings():
+@remote
+def import_db(c, dump_file=None):
     """
-    Copy all settings defined in the environment to the server.
+    Restore the given database dump.
+
+    The dump must be a gzipped SQL dump. If the dump_file parameter is not set,
+    the database will be dumped and retrieved from the remote host.
     """
-    for setting, value in env.settings.items():
-        set_setting(setting, value=value)
+    with open("envdir/DATABASE_URL", "r") as db_credentials_file:
+        db_credentials = db_credentials_file.read()
+    db_credentials_dict = dj_database_url.parse(db_credentials)
 
+    if not is_supported_db_engine(db_credentials_dict["ENGINE"]):
+        raise NotImplementedError(
+            "The import_db task doesn't support your database engine"
+        )
 
-def set_setting(setting_key, value=None, description=None):
-    """
-    Sets the given setting to the given value on the remote server. If the
-    value is not provided, the user will be prompted for it.
+    if dump_file is None:
+        dump_file = fetch_db(c)
 
-    TODO: use the description parameter to display a help text.
-    """
-    if value is None:
-        value = prompt("Please provide value for setting %s: " % setting_key)
+    db_info = {
+        "host": db_credentials_dict["HOST"],
+        "user": db_credentials_dict["USER"],
+        "db": db_credentials_dict["NAME"],
+        "db_dump": dump_file,
+    }
 
-    with cd(os.path.join(get_project_root(), "envdir")):
-        put(StringIO(value), setting_key)
+    c.run(
+        "dropdb -h {host} -U {user} {db}".format(**db_info),
+        env={"PGPASSWORD": db_credentials_dict["PASSWORD"].replace("$", "\$")},
+    )
+    c.run(
+        "createdb -h {host} -U {user} {db}".format(**db_info),
+        env={"PGPASSWORD": db_credentials_dict["PASSWORD"].replace("$", "\$")},
+    )
+    c.run(
+        "gunzip -c {db_dump}|psql -h {host} -U {user} {db}".format(**db_info),
+        env={"PGPASSWORD": db_credentials_dict["PASSWORD"].replace("$", "\$")},
+    )
 
 
 @task
-def bootstrap():
+@remote
+def bootstrap(c):
```

Comment:
It would be nice if the `bootstrap` command could be merged with the `deploy` command, and take care of creating things if they don't exist yet.

Reply:
Now that `deploy` calls `sync_settings`, this part has become problematic:
```python
    required_settings = set(
        [
            "DATABASE_URL",
...
        ]
    )
...
    # Ask for settings that are required but were not set in the parameters
    # file
    for setting in required_settings - set(env_settings.keys()):
        c.conn.set_setting(setting, force=False)
```

This will ask for a `DATABASE_URL` on every deploy, unless we hardcode it in the fabfile.